test/e2e/common/setup-kubestellar.sh

```diff
@@ -13,188 +13,498 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -x # echo so users can understand what is happening
-set -e # exit on error
-
-use_release=false
-KUBESTELLAR_CONTROLLER_MANAGER_VERBOSITY=5
-TRANSPORT_CONTROLLER_VERBOSITY=5
-CLUSTER_SOURCE=kind
-HOSTING_CONTEXT=kind-kubeflex
-
-while [ $# != 0 ]; do
-    case "$1" in
-        (-h|--help)
-            echo "$0 usage: (--released | --kubestellar-controller-manager-verbosity \$num | --transport-controller-verbosity \$num | --env \$kind_or_ocp)*"
-            exit;;
-        (--released)
-            use_release=true;;
-        (--kubestellar-controller-manager-verbosity)
-          if (( $# > 1 )); then
-            KUBESTELLAR_CONTROLLER_MANAGER_VERBOSITY="$2"
-            shift
-          else
-            echo "Missing kubestellar-controller-manager-verbosity value" >&2
-            exit 1;
-          fi;;
-        (--transport-controller-verbosity)
-          if (( $# > 1 )); then
-            TRANSPORT_CONTROLLER_VERBOSITY="$2"
-            shift
-          else
-            echo "Missing transport-controller-verbosity value" >&2
-            exit 1;
-          fi;;
-        (--env)
-          if (( $# < 1 )); then
-            echo "Missing --env value" >&2
-            exit 1
-          fi
-          case "$2" in
-            (kind) CLUSTER_SOURCE=kind;     HOSTING_CONTEXT=kind-kubeflex;;
-            (ocp)  CLUSTER_SOURCE=existing; HOSTING_CONTEXT=kscore;;
-            (*) echo "--env must be given 'kind' or 'ocp'" >&2
-                exit 1;;
-          esac
-          shift;;
-        (*) echo "$0: unrecognized argument/flag '$1'" >&2
+# Enhanced KubeStellar setup script with ArgoCD integration
+set -o errexit          # Exit immediately if any command fails
+set -o nounset          # Exit if any variable is unset
+set -o pipefail         # Fail pipeline if any command in pipeline fails
+set -o errtrace         # Enable error tracing
+shopt -s inherit_errexit  # Ensure subshells inherit error handling
+
+# Configuration defaults
+DEFAULT_ENV="kind"
+DEFAULT_ARGOCD_DOMAIN="argocd.localtest.me"
+DEFAULT_KUBESTELLAR_VERBOSITY=5
+DEFAULT_TRANSPORT_VERBOSITY=5
+DEFAULT_CLUSTER_SOURCE="kind"
+DEFAULT_HOSTING_CONTEXT="kind-kubeflex"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Global variables
+declare -g CLUSTER_SOURCE="$DEFAULT_CLUSTER_SOURCE"
+declare -g HOSTING_CONTEXT="$DEFAULT_HOSTING_CONTEXT"
```

Comment:
Why the explicit `declare -g` for these variables but not for the DEFAULT_XXX variables?

What does the explicit `declare -g` add?